DSharpPlus.CommandsNext/Converters/EntityConverters.cs

```diff
@@ -161,11 +159,10 @@ async Task<Optional<DiscordChannel>> IArgumentConverter<DiscordChannel>.ConvertA
             }
 
             var cs = ctx.Config.CaseSensitive;
-            if (!cs)
-                value = value.ToLowerInvariant();
 
-            var chn = ctx.Guild?.Channels.Values.FirstOrDefault(xc => (cs ? xc.Name : xc.Name.ToLowerInvariant()) == value) ??
-            ctx.Guild?.Threads.Values.FirstOrDefault(xThread => (cs ? xThread.Name : xThread.Name.ToLowerInvariant()) == value);
+            var comparison = cs ? StringComparison.InvariantCulture : StringComparison.InvariantCultureIgnoreCase;
+            var chn = ctx.Guild?.Channels.Values.FirstOrDefault(xc => xc.Name.Equals(value, comparison)) ??
+                      ctx.Guild?.Threads.Values.FirstOrDefault(xThread => xThread.Name.Equals(value, comparison));
```

Comment:
To test this one we need to call command where `DiscordChannel` is one of the parameters. But there are no "good ones" already implemented in test project. So i used `crosspost` command that has DiscordChannel as one of the params. However due to its nature it requires news channel or something. So in this case, error with `403 unauthorized` code will mean that converter successfully converted channel, and `System.ArgumentException` with not suitable overload message will mean that converter failed to convert channel.
`d# crosspost name 939961771798695937`
`d# crosspost naMe 939961771798695937` "name" in these cases is channel name.
![image](https://user-images.githubusercontent.com/50947930/152698360-2bfa7586-45df-4d54-a403-973e73ebfaad.png)